src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -104,17 +106,42 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
             switch (_headerSource)
             {
                 case HeaderSource.OperationLocation when headers.TryGetValue("Operation-Location", out string? operationLocation):
-                    _nextRequestUri = operationLocation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(operationLocation, _startRequestUri);
                     return;
                 case HeaderSource.AzureAsyncOperation when headers.TryGetValue("Azure-AsyncOperation", out string? azureAsyncOperation):
-                    _nextRequestUri = azureAsyncOperation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(azureAsyncOperation, _startRequestUri);
                     return;
                 case HeaderSource.Location when hasLocation:
-                    _nextRequestUri = location!;
+                    _nextRequestUri = AppendOrReplaceApiVersion(location!, _startRequestUri);
                     return;
             }
         }
 
+        private static string AppendOrReplaceApiVersion(string uri, Uri startRequestUri)
+        {
+            NameValueCollection startRequestUriQueries = HttpUtility.ParseQueryString(startRequestUri.Query);
+            var apiVersion = startRequestUriQueries.Get("api-version");
+            if (apiVersion != null)
+            {
+                if (uri.Contains("api-version"))
```

Comment:
Can you provide test cases which have expected input / output to help understand what we are trying to do here.  I suspect we can make this more performant but will need a better idea of the use cases to know that.